Orange/widgets/visualize/utils/widget.py

```diff
@@ -0,0 +1,623 @@
+from collections import Counter, defaultdict
+from xml.sax.saxutils import escape
+from math import log2
+
+import numpy as np
+from scipy import sparse as sp
+
+from AnyQt.QtCore import QSize
+from AnyQt.QtWidgets import QApplication
+
+from Orange.data import Table, ContinuousVariable, Domain, Variable
+from Orange.statistics.util import bincount
+
+from Orange.widgets import gui, report
+from Orange.widgets.settings import (
+    Setting, ContextSetting, DomainContextHandler, SettingProvider
+)
+from Orange.widgets.utils.annotated_data import (
+    create_annotated_table, ANNOTATED_DATA_SIGNAL_NAME, create_groups_table,
+    get_unique_names
+)
+from Orange.widgets.utils.colorpalette import (
+    ColorPaletteGenerator, ContinuousPaletteGenerator, DefaultRGBColors
+)
+from Orange.widgets.utils.plot import OWPlotGUI
+from Orange.widgets.utils.sql import check_sql_input
+from Orange.widgets.visualize.owscatterplotgraph import OWScatterPlotBase
+from Orange.widgets.widget import OWWidget, Input, Output, Msg
+
+
+MAX_CATEGORIES = 11  # maximum number of colors or shapes (including Other)
+MAX_POINTS_IN_TOOLTIP = 5
+
+
+class OWProjectionWidgetBase(OWWidget):
```

Comment:
Names of classes (already in the previous PR, so it's my fault) are not very informative.

Perhaps this could be `OWProjectionWidget` and the derived class, which is currently `OWProjectionWidget` should become `OWDataProjectionWidget`. This reflects the difference - this one can show any projection, including networks, while `OWDataProjectionWidget` has a data signal.